### PR TITLE
Gen3

### PR DESCRIPTION
**Pull request by Vuwij**
Type of PR

- [ ] Code Refactor
- [ ] Bug fix on real robot
- [ ] Bug fix in simulation
- [x] Feature on real robot
- [ ] Feature in simulation
- [ ] Documentation or other improvement

Checklist

- [ ] Only the minimal change is included in this PR the commits are kept minimal
- [ ] Code is readable and documented in the code
- [ ] Minimal tests have been added

### CODE REVIEW

File: amcl [hunk 1]
```diff
@@ -1 +0,0 @@
-external/navigation/amcl/
```
Comment (Vuwij) [inline amcl:1]: undo this

File: .idea/misc.xml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
   <component name="CMakeWorkspace" PROJECT_DIR="$PROJECT_DIR$" />
-  <component name="ProjectRootManager" version="2" project-jdk-name="Python 3.8" project-jdk-type="Python SDK" />
```
Comment (Vuwij) [inline .idea/misc.xml:4]: remove

File: external/.dockerignore [hunk 3]
```diff
@@ -1,6 +0,0 @@
-webots/.*
```
Comment (Vuwij) [inline external/.dockerignore:1]: undo this

File: soccer_pycontrol/src/soccer_pycontrol/soccerbot.py [hunk 4]
```diff
@@ -19,58 +20,70 @@
...
+    RIGHT_LEG_3 = rospy.get_param("~joint_indices/RIGHT_LEG_3", 12)
+    RIGHT_LEG_4 = rospy.get_param("~joint_indices/RIGHT_LEG_4", 13)
+    RIGHT_LEG_5 = rospy.get_param("~joint_indices/RIGHT_LEG_5", 14)
+    RIGHT_LEG_6 = rospy.get_param("~joint_indices/RIGHT_LEG_6", 15)
+    HEAD_1 = rospy.get_param("~joint_indices/HEAD_1", 16)
+    HEAD_2 = rospy.get_param("~joint_indices/HEAD_2", 17)
+    HEAD_CAMERA = rospy.get_param("~joint_indices/HEAD_CAMERA", 18)
+    IMU = rospy.get_param("~joint_indices/IMU", 19)
+
+
```
Comment (Vuwij) [inline soccer_pycontrol/src/soccer_pycontrol/soccerbot.py:68]: Fix joint direction in the URDF or fusion

File: soccer_pycontrol/src/soccer_pycontrol/soccerbot.py [hunk 5]
```diff
@@ -114,17 +128,18 @@ def __init__(self, pose, useFixedBase=False, useCalibration=False):
 
         self.left_foot_init_position = self.get_link_transformation(Links.TORSO, Links.LEFT_LEG_6)
         self.left_foot_init_position[2, 3] = -(self.hip_to_torso[2, 3] + self.walking_hip_height) + self.foot_center_to_floor
+        # self.left_foot_init_position[0, 3] = -self.right_foot_init_position[0, 3]
```
Comment (Vuwij) [inline soccer_pycontrol/src/soccer_pycontrol/soccerbot.py:131]: delete

File: soccer_pycontrol/src/soccer_pycontrol/soccerbot.py [hunk 6]
```diff
@@ -233,7 +248,7 @@ def ready(self):
     def updateRobotConfiguration(self):
         try:
             joint_state = rospy.wait_for_message("joint_states", JointState, timeout=3)
-            self.configuration[0:18] = joint_state.position
+            self.configuration[0:20] = joint_state.position
```
Comment (Vuwij) [inline soccer_pycontrol/src/soccer_pycontrol/soccerbot.py:251]: fix to 18

File: soccer_pycontrol/src/soccer_pycontrol/test_walking.py [hunk 7]
```diff
@@ -35,12 +38,29 @@
 
     rospy.Time = MagicMock()
     joint_state = MagicMock()
-    joint_state.position = [0.0] * 18
+    joint_state.position = [0.0] * 20
     rospy.wait_for_message = MagicMock(return_value=joint_state)
     rospy.loginfo_throttle = lambda a, b: None
-    rospy.get_param = lambda a, b: b
+
```
Comment (Vuwij) [inline soccer_pycontrol/src/soccer_pycontrol/test_walking.py:44]: put this function and robot_model into TestWalking

Comment (Vuwij) [reply soccer_pycontrol/src/soccer_pycontrol/test_walking.py:44]: Instea used parameterized unittests 

https://stackoverflow.com/questions/32899/how-do-you-generate-dynamic-parameterized-unit-tests-in-python

So you can run the same test for all 3 robots

File: soccer_strategy/src/communication/game_controller_receiver.py [hunk 8]
```diff
@@ -47,6 +47,7 @@ def receive_forever(self):
         while not rospy.is_shutdown():
             try:
                 data, peer = self.receiver_socket.recvfrom(GameState.sizeof())
+                print("******************GAMECONTROLLER")
```
Comment (Vuwij) [inline soccer_strategy/src/communication/game_controller_receiver.py:50]: remove

File: soccerbot/launch/soccerbot.launch [hunk 9]
```diff
@@ -14,7 +14,7 @@
 
     <!-- prettier-ignore -->
 
-    <param name="robot_description" if="$(eval robot_model == 'bez1')" command="xacro '$(find bez1_description)/urdf/soccerbot.xacro'
+    <param name="robot_description" if="$(eval robot_model == 'bez1')" command="xacro '$(find bez1_description)/urdf/bez1.xacro'
```
Comment (Vuwij) [inline soccerbot/launch/soccerbot.launch:17]: you need to change this also in many places

**Comment by acrylic-origami** [comment]
Split into two PRs. See #518 and #519

**Comment by acrylic-origami** [comment]
Split into two PRs. See #518 and #519 .

**Pull request closed without merging**